TestCompactFlushQueuedMemTableAndFlushMetrics flaky on windows

https://github.com/cockroachdb/pebble/runs/6203219476?check_suite_focus=true

```
 === RUN   TestCompactFlushQueuedMemTableAndFlushMetrics
    assertion_compare.go:230: 
        	Error Trace:	compaction_test.go:3002
        	            				compaction_test.go:3011
        	Error:      	"50000" is not less than "0"
        	Test:       	TestCompactFlushQueuedMemTableAndFlushMetrics
        	Messages:   	[]
--- FAIL: TestCompactFlushQueuedMemTableAndFlushMetrics (0.01s)
```

--- a/replay/replay_test.go
+++ b/replay/replay_test.go
@@ -507,6 +507,9 @@ func TestCompactionsQuiesce(t *testing.T) {
 			wait := 30 * time.Second
 			if invariants.Enabled {
 				wait = time.Minute
+				if invariants.RaceEnabled {
+					wait = 5 * time.Minute
+				}
 			}
 
 			// The above call to [Wait] should eventually return. [Wait] blocks

